client/me/purchases/purchases-list-in-dataviews/index.tsx

```diff
@@ -69,59 +74,86 @@ function MembershipSubscriptions( {
 	return <MembershipsDataViews memberships={ memberships } />;
 }
 
-function isDataLoading( {
-	isFetchingUserPurchases,
+const PurchasesListDataView: React.FC<
+	PurchasesListProps & PurchasesListConnectedProps & WithStoredPaymentMethodsProps
+> = ( {
 	hasLoadedUserPurchasesFromServer,
-}: {
-	hasLoadedUserPurchasesFromServer: boolean;
-	isFetchingUserPurchases: boolean;
-} ) {
-	if ( isFetchingUserPurchases && ! hasLoadedUserPurchasesFromServer ) {
-		return true;
-	}
-}
-
-function ConciergeBanner( {
+	isFetchingUserPurchases,
+	purchases,
+	subscriptions,
+	sites,
 	nextAppointment,
-	availableSessions,
 	isUserBlocked,
-}: Pick<
-	PurchasesListConnectedProps,
-	'nextAppointment' | 'availableSessions' | 'isUserBlocked'
-> ) {
-	return (
-		<PurchaseListConciergeBanner
-			nextAppointment={ nextAppointment ?? undefined }
-			availableSessions={ availableSessions }
-			isUserBlocked={ isUserBlocked }
-		/>
-	);
-}
+	availableSessions,
+	userId,
+} ) => {
+	const translate = useTranslate();
+	const {
+		data: transferredOwnershipPurchases = [],
+		isLoading,
+		isSuccess: hasLoadedTransferredOwnershipPurchases,
+	} = useGetJetpackTransferredLicensePurchases( { userId } );
+
+	const isDataLoading = useCallback( () => {
+		if (
+			( isFetchingUserPurchases && ! hasLoadedUserPurchasesFromServer ) ||
+			( isLoading && ! hasLoadedTransferredOwnershipPurchases )
+		) {
+			return true;
+		}
+
+		return false;
+	}, [
+		hasLoadedUserPurchasesFromServer,
+		isFetchingUserPurchases,
+		isLoading,
+		hasLoadedTransferredOwnershipPurchases,
+	] );
+
+	const allPurchasesLoaded =
+		hasLoadedUserPurchasesFromServer && hasLoadedTransferredOwnershipPurchases;
+
+	const allPurchases = useMemo( () => {
+		if ( allPurchasesLoaded ) {
+			return [ ...purchases, ...transferredOwnershipPurchases ];
+		}
+		return [];
+	}, [ allPurchasesLoaded, purchases, transferredOwnershipPurchases ] );
+
+	const renderConciergeBanner = () => {
+		return (
+			<PurchaseListConciergeBanner
+				nextAppointment={ nextAppointment ?? undefined }
+				availableSessions={ availableSessions }
+				isUserBlocked={ isUserBlocked }
+			/>
+		);
+	};
 
-function PurchasesListDataView(
-	props: PurchasesListProps &
-		PurchasesListConnectedProps &
-		WithStoredPaymentMethodsProps &
-		LocalizeProps
-) {
-	const { purchases, sites, translate, subscriptions } = props;
 	const commonEventProps = { context: 'me' };
 	let content;
 
-	if (
-		isDataLoading( {
-			isFetchingUserPurchases: props.isFetchingUserPurchases,
-			hasLoadedUserPurchasesFromServer: props.hasLoadedUserPurchasesFromServer,
-		} )
-	) {
+	if ( isDataLoading() ) {
 		content = <PurchasesSite isPlaceholder />;
 	}
 
-	if ( purchases && purchases.length ) {
-		content = <PurchasesDataViews purchases={ purchases } sites={ sites } />;
+	if ( allPurchases && allPurchasesLoaded && allPurchases.length ) {
```

Comment:
Not a big deal, more of a nitpick, but allPurchases seem to be always defined, and allPurchases.length > 0 is only when allPurchasesLoaded is true. so from all these checks, we could just leave allPurchases.length.